src/main/java/namor/template/calculator/RPNcalculator.java

```diff
@@ -1,66 +1,117 @@
 package namor.template.calculator;
 
 import namor.template.stack.ArrayStack;
+import namor.template.stack.MapBasedStack;
+import namor.template.stack.Stack;
 
-public class RPNcalculator {
+public class RPNcalculator<T extends Number>{
+    private final Class<T> type;
+    private Stack<T> numbers;
 
+    public Stack<T> getNumbers() {
+        return numbers;
+    }
+
+    public RPNcalculator(Class<T> type, Class stack) {
+        if (stack.equals(ArrayStack.class)){
+            numbers = new ArrayStack<>();
+        } else if (stack.equals(MapBasedStack.class)){
+            numbers = new MapBasedStack<>();
+        }
+        this.type = type;
+    }
 
-    public int calculate(String arithmeticExpression) {
-        ArrayStack numbers = new ArrayStack();
+    private Class<T> getMyType() {
+        return this.type;
+    }
+
+    public T calculate(String arithmeticExpression) {
 
         for (String operationElement : arithmeticExpression.split(" ")) {
             if (Sign.isSign(operationElement)) {
                 doArithmeticOperation(operationElement, numbers);
             } else {
-                numbers.push(Integer.parseInt(operationElement));
+                convertStringToNumber(getMyType(), numbers, operationElement);
             }
         }
         return numbers.pop();
     }
 
-    private void doArithmeticOperation(String arithmeticOperator, ArrayStack numbers) {
+    private void doArithmeticOperation(String arithmeticOperator, Stack<T> numbers) {
         switch (Sign.getValue(arithmeticOperator)) {
             case ADDITION:
-                numbers.push(addition(numbers));
+                numbers.push(addition(numbers.pop(), numbers.pop()));
                 break;
             case SUBTRACTION:
-                numbers.push(subtraction(numbers));
+                numbers.push(subtraction(numbers.pop(), numbers.pop()));
                 break;
             case MULTIPLICATION:
-                numbers.push(multiplication(numbers));
+                numbers.push(multiplication(numbers.pop(), numbers.pop()));
                 break;
             case DIVISION:
-                numbers.push(division(numbers));
+                numbers.push(division(numbers.pop(), numbers.pop()));
                 break;
         }
     }
 
-    private final int addition(ArrayStack numbers) {
-        final int secondNumber = numbers.pop();
-        final int firstNumber = numbers.pop();
-
-        return firstNumber + secondNumber;
+    private void convertStringToNumber(Class<T> type, Stack<T> numbers, String operationElement) {
```

Comment:
Get rid of converters. First of all always think that things what you want to do can already be implemented. To convert Strings to Numbers and vice versa use String.valueOf(), Double.valueOf, Integer.valueOf etc. In other words java have already all converters